packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -0,0 +1,54 @@
+import vscode from "vscode";
+
+const COMMAND = "CodeActionProvider.Custom.OpenUrl";
+
+export function createCodeActionProvider() {
+  return vscode.languages.registerCodeActionsProvider(
+    "typespec",
+    new ExtensionCodeActionProvider(),
+    {
+      providedCodeActionKinds: ExtensionCodeActionProvider.providedCodeActionKinds,
+    },
+  );
+}
+
+export function createCommandOpenUrl() {
```

Comment:
the command is not necessary coupling with CodeAction. please create a vscode-command file and move this there